src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -135,6 +139,200 @@ void RaiseSecond()
             VisualStudio.Editor.Verify.TextContains("Second?.");
         }
 
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.EditorConfig)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        [WorkItem(15003, "https://github.com/dotnet/roslyn/issues/15003")]
+        [WorkItem(19089, "https://github.com/dotnet/roslyn/issues/19089")]
+        public void ApplyEditorConfigAndFixAllOccurrences()
+        {
+            var markup = @"
+class C
+{
+    public int X1
+    {
+        get
+        {
+            $$return 3;
+        }
+    }
+
+    public int Y1 => 5;
+
+    public int X2
+    {
+        get
+        {
+            return 3;
+        }
+    }
+
+    public int Y2 => 5;
+}";
+            var expectedText = @"
+class C
+{
+    public int X1 => 3;
+
+    public int Y1 => 5;
+
+    public int X2 => 3;
+
+    public int Y2 => 5;
+}";
+
+            // CodingConventions only sends notifications if a file is open for all directories in the project
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), @"Properties\AssemblyInfo.cs");
+
+            // Switch back to the main document we'll be editing
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "Class1.cs");
+
+            /*
+             * The first portion of this test adds a .editorconfig file to configure the analyzer behavior, and verifies
+             * that diagnostics appear automatically in response to the newly-created file. A fix all operation is
+             * applied, and the result is verified against the expected outcome for the .editorconfig style.
+             */
+
+            MarkupTestFile.GetSpans(markup, out var text, out ImmutableArray<TextSpan> spans);
+            SetUpEditor(markup);
+            VisualStudio.WaitForApplicationIdle(CancellationToken.None);
+            VisualStudio.Editor.Verify.CodeActionsNotShowing();
+
+            var editorConfig = @"root = true
+
+[*.cs]
+csharp_style_expression_bodied_properties = true:warning
+";
+
+            VisualStudio.SolutionExplorer.AddFile(new ProjectUtils.Project(ProjectName), ".editorconfig", editorConfig, open: false);
+
+            // We have no way of waiting on the CodingConventions library to send its file system changed notifications
+            Thread.Sleep(TimeSpan.FromSeconds(1));
```

Comment:
wait for editor config update events from fired? there is WiatUntil thing I added.

Reply:
if race of that is problem due to our Iasync listener not having a way to see whether something ever happened.

you can either build one into it. or do WaitUntil in other thread with manualreset and then call addfile ".editcofnig", and wait for the manualreset to fire.

Reply:
CodingConventions responds to the file change by firing an asynchronous event outside Roslyn's tracking abilities. I would prefer to have a deterministic trigger here but it wasn't clear how to do that.

Reply:
you can do 
ManualResetEvent e = ...
Task.Run(() => editorconfigListener.WaitUntilConditionIsMetAsync(tokens => tokens.Any(t => check token is one for UpdateProject)).ContinueWith(_ => e.Set, ... );

VS.SolutionExplorer.AddFile(.... "editorconfig");

e.WaitOne();

something like that.

Reply:
that above will make sure workspace events is pending which you listen below. or you can add EditorConfigOptionProvider in the waiters below.